Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -2336,6 +2408,14 @@ void Parser::consume_or_insert_semicolon()
 static constexpr AK::Array<StringView, 38> reserved_words = { "await", "break", "case", "catch", "class", "const", "continue", "debugger", "default", "delete", "do", "else", "enum", "export", "extends", "false", "finally", "for", "function", "if", "import", "in", "instanceof", "new", "null", "return", "super", "switch", "this", "throw", "true", "try", "typeof", "var", "void", "while", "with", "yield" };
 static constexpr AK::Array<StringView, 9> strict_reserved_words = { "implements", "interface", "let", "package", "private", "protected", "public", "static", "yield" };
 
+Token Parser::consume_identifier_name()
+{
+    if (!match_identifier_name())
+        expected("IdentifierName");
+
+    return consume();
```

Comment:
Should `TokenType::Identifier` be added as a parameter here?
Otherwise the reserved words check is skipped.

Reply:
- This method consumes identifier names, not just identifiers (e.g. `let` in `(let[4])`)
- That check is redundant AFAICT, the lexer never produces `TokenType::Identifier` for keywords

Reply:
That check is not redundant, it fixed 3 test262 test cases if i remember correctly.

Reply:
Then I'd start questioning "which keywords are being treated as identifiers in the lexer".

Reply:
The answer to that is, checking against `reserved_words` is entirely redundant (they all occur in `s_keywords` - see Lexer.cpp:29-68 and 474), but checking against `strict_reserved_words` is not.
I'll move some stuff around to make it match the spec more closely.